module/private_quarters/interact.py

```diff
@@ -0,0 +1,274 @@
+from module.base.timer import Timer
+from module.base.utils import random_rectangle_vector
+from module.handler.assets import POPUP_CANCEL
+from module.logger import logger
+from module.private_quarters.assets import *
+from module.ui.page import page_private_quarters
+from module.ui.ui import UI
+
+
+class PQInteract(UI):
+    # Key: str, target ship name
+    # Value: list[Button], button instances
+    #        (Room_Entrance, Page_Locale)
+    available_targets = {
+        'anchorage': (PRIVATE_QUARTERS_SHIP_ANCHORAGE, PRIVATE_QUARTERS_PAGE_LOCALE_BEACH),
+        'noshiro': (PRIVATE_QUARTERS_SHIP_NOSHIRO, PRIVATE_QUARTERS_PAGE_LOCALE_BEACH),
+        'sirius': (PRIVATE_QUARTERS_SHIP_SIRIUS, PRIVATE_QUARTERS_PAGE_LOCALE_BEACH),
+        'new_jersey': (PRIVATE_QUARTERS_SHIP_NEW_JERSEY, PRIVATE_QUARTERS_PAGE_LOCALE_LOFT),
+    }
+
+    def _pq_target_appear(self):
+        """
+        Callable wrapper to validate target's appearance
+        offset=(100, 100) detectable for anchorage, noshiro, sirus, and new_jersey
+        When more ships added may need to adjust or capture specific bubble position per
+        ship, can use the available_targets to store similarly into tuples instead
+        """
+        settle_timer = Timer(1.5, count=3).start()
+        skip_first_screenshot = True
+        while 1:
+            if skip_first_screenshot:
+                skip_first_screenshot = False
+            else:
+                self.device.screenshot()
+
+            # End, success
+            if self.appear(PRIVATE_QUARTERS_ROOM_TARGET_CHECK_1, offset=(100, 100)):
+                return True
+            if self.appear(PRIVATE_QUARTERS_ROOM_TARGET_CHECK_2, offset=(100, 100)):
+                return True
+
+            # End, failed expired wait time
+            if settle_timer.reached():
+                return False
+
+            # Factor in couple drag up actions to
+            # counter odd default distance/zoom on target
+            p1, p2 = random_rectangle_vector(
+                (0, -30), box=PRIVATE_QUARTERS_ROOM_SAFE_CLICK_AREA.area, random_range=(-10, -10, 10, 10), padding=5)
+            self.device.drag(p1, p2, segments=2, shake=(0, 25), point_random=(0, 0, 0, 0), shake_random=(0, -5, 0, 5))
+
+    def _pq_goto_room_seek(self, target_ship):
+        """
+        Execute seek room routine
+
+        Args:
+            target_ship (str):
+
+        Returns:
+            bool
+        """
+        target_title = target_ship.title().replace('_', ' ')
+        page_btn = self.available_targets[target_ship][1]
+        logger.hr(f'Seek {target_title}\'s Page', level=2)
```

Comment:
Possible KeyError and IndexError in `self.available_targets[target_ship][1]`, log and skip if input ship is invalid